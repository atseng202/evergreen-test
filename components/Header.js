// import logo from "./logo.svg";
import STYLES from "../styles/Header.module.scss";

const getClassName = (className) => STYLES[className] || "UNKNOWN";

const Header = () => (
  <header className={getClassName("Header")}>
    <a href="/">
      <span className={getClassName("Header__hidden-text")}>Skyscanner</span>
      <img
        src="/logo.svg"
        className={getClassName("Header__logo-image")}
        alt="Skyscanner"
      />
    </a>
  </header>
);

export default Header;
