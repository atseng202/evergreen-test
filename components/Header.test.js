import React from "react";
import { render } from "@testing-library/react";
import Header from "./Header";

describe("Header smoke and snapshot tests", () => {
  it("should render correctly", () => {
    render(<Header />);
  });

  it("matches Header snapshot", () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
