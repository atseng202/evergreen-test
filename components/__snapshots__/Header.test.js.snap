// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header smoke and snapshot tests matches Header snapshot 1`] = `
<DocumentFragment>
  <header
    class="Header"
  >
    <a
      href="/"
    >
      <span
        class="Header__hidden-text"
      >
        Skyscanner
      </span>
      <img
        alt="Skyscanner"
        class="Header__logo-image"
        src="/logo.svg"
      />
    </a>
  </header>
</DocumentFragment>
`;
